### ISSUE TITLE
jruby 1.7.16 and 4.2.0.beta2 ActiveSupport::Duration issue

### ISSUE THREAD

**Issue by petergoldstein** [jruby/jruby#2041]
In Rails 4.2 a new class, ActiveSupport::Duration was introduced.  So where previously an expression like:

```
`7.days`
```

would have type Fixnum, it now has type `ActiveSupport::Duration`.  As you'd expect, calling `to_i` on the `ActiveSupport::Duration` object gives you the old Fixnum value.

With JRuby 1.7.16 I'm seeing an issue using `pack` with arrays containing `ActiveSupport::Duration` objects.  Specifically, if I do the following:

```
require 'active_support/core_ext/numeric/time'
[7.days].pack('N')
```

I get an error that looks like this:

```
TypeError: can't convert ActiveSupport::Duration into Integer
  from org/jruby/RubyArray.java:4125:in `pack'
  from (irb):2:in `evaluate'
  from org/jruby/RubyKernel.java:1101:in `eval'
  from org/jruby/RubyKernel.java:1501:in `loop'
  from org/jruby/RubyKernel.java:1264:in `catch'
  from org/jruby/RubyKernel.java:1264:in `catch'
```

Now one can certainly argue that this isn't a JRuby problem, it's a Rails issue.  But:
1. MRI handles this transparently
2. This has a wide impact as it comes up whenever code sends a value like `7.days` to be serialized.  So it impacts ActiveRecord, Dalli, etc.

So ideally it would be nice if JRuby handled this transparently.  Any thoughts?

**Comment by rtyler** [comment]
This smells like it might be related to #1856

**Comment by kares** [comment]
this bug has been fixed, somewhere along the line of JRuby 1.7.x releases : 
```
jruby-1.7.25 :001 > require 'active_support/core_ext/numeric/time'
 => true 
jruby-1.7.25 :002 > ActiveSupport::VERSION::STRING
 => "4.2.8" 
jruby-1.7.25 :005 > 7.days.class
 => ActiveSupport::Duration 
jruby-1.7.25 :006 > [7.days].pack('N')
 => "\x00\t:\x80" 

```
